### PR TITLE
Added terraform files

### PR DESCRIPTION
**Pull request by LeonardoPalmeiro**
### JIRA link ###
ROC-9706

### Change description ###
Added terraform files for infrastructure

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by laarmada — APPROVED** [review]
all correct

File: infrastructure/state.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  backend "azurerm" {}
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=2.49.0"
```
Comment (taomoh) [inline infrastructure/state.tf:7]: This is behind the version referred to in the good example: https://github.com/hmcts/draft-store/tree/master/infrastructure

File: infrastructure/output.tf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+output "env" {
+  value = "${var.env}"
+}
```
Comment (taomoh) [inline infrastructure/output.tf:3]: @mounikahmcts thinks this file isn't required

File: infrastructure/main.tf [hunk 3]
```diff
@@ -0,0 +1,86 @@
+provider "azurerm" {
+ features {}
+}
+
+locals {
+  vaultName = "${var.product}-${var.env}"
+}
+
+data "azurerm_key_vault" "cmc_key_vault" {
```
Comment (taomoh) [inline infrastructure/main.tf:9]: change all `cmc_` prefixes to `civil_`

Comment (taomoh) [reply infrastructure/main.tf:9]: At this stage we may not need all the keys copied from CMC. We can also ask platop to confirm what is the minimum set of key that have to be defined for a skeleton application

File: infrastructure/output.tf (hunk 2 above)
Comment (taomoh) [reply infrastructure/output.tf]: @LeonardoPalmeiro , can you please double-check with @mounikahmcts if this file is required? if it's not please remove it

File: infrastructure/main.tf [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  product             = var.product
+  env                 = var.env
+  tenant_id           = var.tenant_id
+  object_id           = var.jenkins_AAD_objectId
+  resource_group_name = azurerm_resource_group.rg.name
+  product_group_name          = "DTS Civil"
+  common_tags                 = var.common_tags
+}
+
+resource "azurerm_key_vault_secret" "POSTGRES-USER-CIVIL" {
```
Comment (mounikahmcts) [inline infrastructure/main.tf:26]: Quite not sure why do we need postgress here

File: infrastructure/output.tf (hunk 2 above)
Comment (LeonardoPalmeiro) [reply infrastructure/output.tf]: all project has this file for now I'll keep it.

File: infrastructure/main.tf (hunk 4 above)
Comment (LeonardoPalmeiro) [reply infrastructure/main.tf:26]: you're right we don't use this I just added for test propose I'll remove in the next PR.

**Review by taomoh — APPROVED** [review]

File: infrastructure/variables.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "location" {
+  default = "UK South"
+}
+
+variable "env" {}
+
+variable "ilbIp" {}
+
+
```
Comment (timja) [inline infrastructure/variables.tf:19]:
```suggestion
```

File: infrastructure/state.tf [hunk 6]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  backend "azurerm" {}
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "2.50"
```
Comment (timja) [inline infrastructure/state.tf:7]: always start with the latest version

File: infrastructure/.terraform-version [hunk 7]
```diff
@@ -0,0 +1 @@
+0.13.2
```
Comment (timja) [inline infrastructure/.terraform-version:1]: use the latest version

**Review by timja — CHANGES_REQUESTED** [review]
What do you need infra wise here? Does this frontend application need infrastructure? (as you aren't adding any in this PR)

File: infrastructure/state.tf (hunk 6 above)
Comment (LeonardoPalmeiro) [reply infrastructure/state.tf:7]: I used from drat-store.
The last one from [registry](https://registry.terraform.io/providers/hashicorp/azurerm/latest) is 2.84.0, then I'll this one.

File: infrastructure/.terraform-version (hunk 7 above)
Comment (LeonardoPalmeiro) [reply infrastructure/.terraform-version:1]: I'll use the 1.0.11, is there ok?

**Comment by LeonardoPalmeiro** [comment]
> What do you need infra wise here? Does this frontend application need infrastructure? (as you aren't adding any in this PR)

for now we don't need but in the next stages we will.

File: infrastructure/variables.tf [hunk 8]
```diff
@@ -0,0 +1,38 @@
+// Infrastructural variables
+variable "product" {}
+
+variable "component" {}
+
+
+variable "microservice" {
+  default = "citizen-ui"
+}
+
```
Comment (timja) [inline infrastructure/variables.tf:10]: this is not needed either
```suggestion
```

File: infrastructure/variables.tf [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+variable "node_env" {
+  default = "production"
+}
+
+variable "appinsights_instrumentation_key" {
+  description = "Instrumentation key of the App Insights instance this webapp should use. Module will create own App Insights resource if this is not provided"
+  default     = ""
+}
+
```
Comment (timja) [inline infrastructure/variables.tf:35]:
```suggestion
```

File: infrastructure/.terraform-version (hunk 7 above)
Comment (timja) [reply infrastructure/.terraform-version:1]: yup

**Review by timja — COMMENTED** [review]
better can you clean these up too please

File: infrastructure/.terraform-version (hunk 7 above)
Comment (timja) [reply infrastructure/.terraform-version:1]:
```suggestion
1.0.11
```

**Review by timja — APPROVED** [review]
can you update terraform-version file as well, once done then this is ok

**Comment by timja** [comment]
> for now we don't need but in the next stages we will.

what are you planning to add?

File: infrastructure/.terraform-version (hunk 7 above)
Comment (LeonardoPalmeiro) [reply infrastructure/.terraform-version:1]: is in this commit https://github.com/hmcts/civil-citizen-ui/pull/26/commits/02d0f6fa8be1ad945

**Comment by taomoh** [comment]
> > for now we don't need but in the next stages we will.
> 
> what are you planning to add (out of interest)?

@timja we anticipate we will be adding key vault resources like secrets and application variables, if by infrastructure you mean items like a database for example, then we don't anticipate this will be needed

**Review by laarmada — APPROVED** [review]

File: infrastructure/main.tf [hunk 10]
```diff
@@ -0,0 +1,10 @@
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_key_vault" "civil_key_vault" {
+  name = "${var.product}-${var.env}"
+  resource_group_name = "${var.product}-service-${var.env}"
```
Comment (taomoh) [inline infrastructure/main.tf:7]: @LeonardoPalmeiro, this tells me civil-citizen-ui component is reusing another existing component's (civil-service) resource group: civil-service-<env>. Not sure this is sensible. Not a big deal just now, but we will probably need to revisit this in the future so civil-citizen-ui component has its own resource group: civil-citizen-ui-<env>
@timja , would you agree?

**Review by timja — CHANGES_REQUESTED** [review]
> > > for now we don't need but in the next stages we will.
> > 
> > 
> > what are you planning to add (out of interest)?
> 
> @timja we anticipate we will be adding key vault resources like secrets and application variables, if by infrastructure you mean items like a database for example, then we don't anticipate this will be needed

You won't need any of these in a frontend repo normally (generally the only service frontend apps deploy is redis).

Don't anticipate when adding something like this, it will add time to your pipeline unecessarily.

Either you'll need infra or you won't.

But better to add in a just in time manner.

File: infrastructure/main.tf (hunk 10 above)
Comment (timja) [reply infrastructure/main.tf:7]: i'm not sure exactly how your project is setup but normally a set of [components](https://hmcts.github.io/glossary/#component) will use a [products](https://hmcts.github.io/glossary/#product) infrastrucutre, for shared components anyway like key vaults.

**Review by timja — CHANGES_REQUESTED** [review]
not sure why you are updating this, you aren't using the terraform at all

**Review by taomoh — APPROVED** [review]
LGTM

**Comment by LeonardoPalmeiro** [comment]
> 



> not sure why you are updating this, you aren't using the terraform at all

we created a shared infrastructure for civil, now we will use that one.

**Review by timja — CHANGES_REQUESTED** [review]
This PR does nothing, your application will read secrets via CSI driver https://hmcts.github.io/ways-of-working/new-component/secrets-management.html#secrets-management

not via terraform

**Pull request closed without merging**